appsdk/configurable.go

```diff
@@ -423,6 +424,22 @@ func (dynamic AppFunctionsSDKConfigurable) SetOutputData() appcontext.AppFunctio
 	return transform.SetOutputData
 }
 
+// SetOutputDataWithContentType sets the output data (with content type) to that passed in from the previous function.
+// It will return an error and stop the pipeline if data passed in is not of type []byte, string or json.Mashaler
+// This function is a configuration function and returns a function pointer.
+// TODO: rework this for v2 to collapse to one function (above)
+func (dynamic AppFunctionsSDKConfigurable) SetOutputDataWithContentType(parameters map[string]string) appcontext.AppFunction {
+	transform := transforms.OutputData{}
+
+	value, ok := parameters[ResponseContentType]
+	if !ok || len(value) == 0 {
```

Comment:
So the ResponseContentType is required.

Reply:
When using function SetOutputDataWithContentType() the content type is required. If you use that function and don't utilize the content-type parameter, then it acts just like the SetOutputData() function above it. Our thinking here was that if someone is explicitly using SetOutputDataWithContentType(), then it should be required - if not, they should use SetOutputData().

Also noteworthy, we realize this is a bit janky, but to introduce the functionality without introducing a breaking change, we can't modify the SetOutputData() function signature. Recommend that we collapse those two functions when work begins on v2 implementation.

Reply:
Got it.